pgx-examples/aggregate/src/lib.rs

```diff
@@ -0,0 +1,122 @@
+use pgx::*;
+use std::{
+    str::FromStr,
+    ffi::CStr,
+};
+
+pg_module_magic!();
+
+
+use serde::{Deserialize, Serialize};
+
+
+#[derive(Copy, Clone, PostgresType)]
+#[pgvarlena_inoutfuncs]
+pub struct IntegerAvgState {
+    sum: i32,
+    n: i32,
+}
+
+impl PgVarlenaInOutFuncs for IntegerAvgState {
+    fn input(input: &CStr) -> PgVarlena<Self> {
+        let mut result = PgVarlena::<Self>::new();
+
+        let mut split = input.to_bytes().split(|b| *b == b',');
+        let sum = split.next().map(|value| 
+            i32::from_str(unsafe { std::str::from_utf8_unchecked(value) }).expect("invalid i32")
+        ).expect("expected sum");
+        let n = split.next().map(|value| 
+            i32::from_str(unsafe { std::str::from_utf8_unchecked(value) }).expect("invalid i32")
+        ).expect("expected n");
+
+        result.sum = sum;
+        result.n = n;
+
+        result
+    }
+    fn output(&self, buffer: &mut StringInfo) {
+        buffer.push_str(&self.sum.to_string());
+        buffer.push(',');
+        buffer.push_str(&self.n.to_string());
```

Comment:
Personal preference only. Slightly easier to skim read.

```suggestion
        buffer.push_str(&format!("{},{}"), self.sum, self.n);
```

Reply:
That would be doing allocations in what might be a hot path for Postgres. The idea of StringBuffer is to save that.

Reply:
One of the reasons I considered using format was that it would only involve 1 allocation rather than 2 (each call to `.to_string()`).

Reply:
Fair. Push it